ofrak_patch_maker/ofrak_patch_maker/toolchain/abstract.py

```diff
@@ -357,11 +357,16 @@ def link(self, o_files: List[str], exec_path: str, script: str = None):
     def linker_include_filter(symbol_name: str) -> bool:
         return "." in symbol_name or "_DYNAMIC" in symbol_name
 
-    def keep_section(self, section_name: str) -> bool:
-        if self._config.separate_data_sections:
-            raise NotImplementedError("you must override keep_section() in your Toolchain sublass")
```

Comment:
Out of curiosity, do you know why this was done this way in the first place?

Reply:
I believe it is because flexible keep sections was only allowed because of the `separate_data_sections` option, which would have to be implemented in a toolchain-dependent way. Each toolchain could use a different flag and generate its own pattern of data section names. But it turns out the pattern is pretty standard for the two major families we have (LLVM and GNU). Overriding this is still possible if a toolchain breaks the pattern.